components/example-notebook-servers/base/Dockerfile

```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:20.04
+
+ENV NB_USER jovyan
+ENV NB_UID 1000
+ENV NB_PREFIX /
+ENV HOME /home/$NB_USER
+
+# set version for s6 overlay
+ARG OVERLAY_VERSION="v2.2.0.3"
+ARG OVERLAY_ARCH="amd64"
+
+# Install -- Linux Dependencies
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get -yq update \
+    && apt-get -yq install --no-install-recommends \
+    bash \
+    bzip2 \
+    ca-certificates \
+    curl \
+    git \
+    gnupg \
+    locales \
+    lsb-release \
+    nano \
+    software-properties-common \
+    tzdata \
+    unzip \
+    vim \
+    wget \
+    zip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Set locale
+RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
+    locale-gen
+ENV LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8 \
+    SHELL=/bin/bash
+
+# add s6 overlay
+ADD https://github.com/just-containers/s6-overlay/releases/download/${OVERLAY_VERSION}/s6-overlay-${OVERLAY_ARCH}-installer /tmp/
```

Comment:
@DavidSpek just to have it documented, why do you install s6 in the base image?

Reply:
S6 is the init system that starts the processes in the container. It is an alternative to `Tini`, and is used in all the upstream containers. 
Very short description of why S6 over Tini. S6 can properly handle multiple processes in a container (Tini is only really built to handle 1 process), and it allows you to easily run scripts on container start and such.

Reply:
Specifically this I am planning to write detailed documentation for as S6 is very powerful.

Reply:
Thanks for the info!

> Specifically this I am planning to write detailed documentation for as S6 is very powerful.

That's awesome! This will also be needed in order to explain users our reasons and pain points we want to avoid by using it.

So the plan is for every following Notebook server right?

And lastly
> It is an alternative to Tini, and is used in all the upstream containers.

Do you have a short list of projects using it readily available? I believe this will also help users understand why it's useful

Reply:
Yes, RStudio and VS-Code (Code-Server) were already using S6, so I moved Jupyter to use that as well (first I was using tini there). So now every notebook downstream uses S6 for the init system. 

The Rocker project uses S6 for their init system. https://github.com/rocker-org/rocker-versioned2
Another popular container image for Code-Server with 10M+ pull on dockerhub is using S6 for the init system. https://hub.docker.com/r/linuxserver/code-server

I can take a bit of time later on to come up with a small list that we can actually use in the docs if you like.

Reply:
It's OK, just these are more than fine for now. 
I'm good, again thanks for the info!

[ feel free to resolve this ]